modules/blocks/process.js

```diff
@@ -165,33 +165,73 @@ Process.prototype.loadBlocksOffset = function (limit, offset, verify, cb) {
 				}
 
 				library.logger.debug('Processing block', block.id);
+
 				if (verify && block.id !== library.genesisblock.block.id) {
-					// Sanity check of the block, if values are coherent.
-					// No access to database.
-					var check = modules.blocks.verify.verifyBlock(block);
-
-					if (!check.verified) {
-						library.logger.error(['Block', block.id, 'verification failed'].join(' '), check.errors.join(', '));
-						// Return first error from checks
-						return setImmediate(cb, check.errors[0]);
-					}
-				}
-				if (block.id === library.genesisblock.block.id) {
-					modules.blocks.chain.applyGenesisBlock(block, cb);
+					async.series({
+						normalizeBlock: function (seriesCb) {
+							// normalizeBlock
+							try {
+								block = library.logic.block.objectNormalize(block);
+							} catch (err) {
+								return setImmediate(seriesCb, err);
+							}
+
+							return setImmediate(seriesCb);
+						},
+						verifyBlock: function (seriesCb) {
+							// Sanity check of the block, if values are coherent.
+							// No access to database
+							modules.blocks.verify.verifyBlock(block, function (err) {
+								if (err) {
+									library.logger.error(['Block', block.id, 'verification failed'].join(' '), err);
+									return setImmediate(seriesCb, err);
+								}
+
+								return setImmediate(seriesCb);
+							});
+						},
+						validateBlockSlot: function (seriesCb) {
+							// Check if block was generated by the right active delagate. Otherwise, fork 3.
+							// DATABASE: Read only to mem_accounts to extract active delegate list
+							modules.delegates.validateBlockSlot(block, function (err) {
+								if (err) {
+									// Fork: Delegate does not match calculated slot.
+									modules.delegates.fork(block, 3);
+									return setImmediate(seriesCb, err);
+								} else {
+									return setImmediate(seriesCb);
+								}
+							});
+						}
+					}, function (err) {
+						if (err) {
+							return setImmediate(cb, err);
+						} else {
+							// Apply block - broadcast: false, saveBlock: false
+							modules.blocks.chain.applyBlock(block, false, cb, false);
+
+							// Update last block
+							modules.blocks.lastBlock.set(block);
```

Comment:
Good catch!, `modules.blocks.lastBlock.set(block);` is been called within `modules.blocks.chain.applyBlock`, so I will delete it from process. Also `cb` needs to be transformed into function